application/common/components/adapters/SecureUserIdStore.php

```diff
@@ -0,0 +1,164 @@
+<?php
+namespace Sil\Idp\IdSync\common\components\adapters;
+
+use Exception;
+use GuzzleHttp\Client;
+use InvalidArgumentException;
+use Sil\Idp\IdSync\common\components\IdStoreBase;
+use Sil\Idp\IdSync\common\models\User;
+use yii\helpers\Json;
+
+class SecureUserIdStore extends IdStoreBase
+{
+    public $apiUrl = null;
+    public $apiKey = null;
+    public $apiSecret = null;
+
+    public $timeout = 45; // Timeout in seconds (per call to ID Store API).
+
+    protected $httpClient = null;
+
+    public function init()
+    {
+        $requiredProperties = [
+            'apiUrl',
+            'apiKey',
+            'apiSecret',
+        ];
+        foreach ($requiredProperties as $requiredProperty) {
+            if (empty($this->$requiredProperty)) {
+                throw new InvalidArgumentException(sprintf(
+                    'No %s was provided.',
+                    $requiredProperty
+                ), 1642083101);
+            }
+        }
+
+        parent::init();
+    }
+
+    public static function getFieldNameMap(): array
+    {
+        return [
+            'employee_number' => User::EMPLOYEE_ID,
+            'first_name' => User::FIRST_NAME,
+            'last_name' => User::LAST_NAME,
+            'display_name' => User::DISPLAY_NAME,
+            'email' => User::EMAIL,
+            'username' => User::USERNAME,
+            'locked' => User::LOCKED,
+            'manager_email' => User::MANAGER_EMAIL,
+        ];
+    }
+
+    /**
+     * Get the specified user's information. Note that inactive users will be
+     * treated as non-existent users.
+     *
+     * @param string $employeeId The Employee ID.
+     * @return User|null Information about the specified user, or null if no
+     *     such active user was found.
+     * @throws Exception
+     */
+    public function getActiveUser(string $employeeId)
+    {
+        $allUsers = $this->getAllActiveUsers();
+        foreach ($allUsers as $user) {
+            if ((string)$user->getEmployeeId() === (string)$employeeId) {
+                return $user;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Get a list of users' information (containing at least an Employee ID) for
+     * all users changed since the specified time.
+     *
+     * @param int $unixTimestamp The time (as a UNIX timestamp).
+     * @return User[]
+     * @throws Exception
+     */
+    public function getUsersChangedSince(int $unixTimestamp)
+    {
+        throw new Exception(__FUNCTION__ . ' not yet implemented');
+    }
+
+    /**
+     * Get information about each of the (active) users.
+     *
+     * @return User[] A list of Users.
+     * @throws Exception
+     */
+    public function getAllActiveUsers()
+    {
+        $client = $this->getHttpClient();
+
+        $api_sig = hash_hmac('sha256', time().$this->apiKey, $this->apiSecret);
```

Comment:
Is this right? `time().$this->apiKey`

Reply:
Yep. Concatenation of the time and the api key.

Reply:
Oh, haha, I was reading the `.` as if it was Go. I think maybe spaces on each side of the `.` are more standard.
```suggestion
        $api_sig = hash_hmac('sha256', time() . $this->apiKey, $this->apiSecret);
```